server/s3_get_object_action.cc

```diff
@@ -825,6 +839,10 @@ void S3GetObjectAction::send_data_to_client() {
     request->set_out_header_value("Accept-Ranges", "bytes");
     request->set_out_header_value(
         "Content-Length", std::to_string(get_requested_content_length()));
+    if (bucket_metadata->get_bucket_versioning_status() != "Unversioned") {
+      request->set_out_header_value("x-amz-version-id",
```

Comment:
low priority change

can we use 
set_version_id()

"x-amz-version-id" this field related to object metadata 
and it will be good if we can hide it from action

Reply:
I think it should be fine to do it here as we are setting all other metadata attribute values here.